diff --git a/internal/cache/position.go b/internal/cache/position.go
new file mode 100644
index 000000000..a9f37c841
--- /dev/null
+++ b/internal/cache/position.go
@@ -0,0 +1,54 @@
+package cache
+
+import (
+	"go/token"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+// RelativePosition creates a [token.Position] with a filename relative to the module directory.
+func RelativePosition(mod *packages.Module, pos token.Position) token.Position {
+	pos.Filename, _ = toRelativePath(mod, pos.Filename)
+
+	return pos
+}
+
+// AbsolutionPosition creates the absolute path base on the module directory.
+func AbsolutionPosition(mod *packages.Module, pos token.Position) token.Position {
+	pos.Filename = toAbsolutePath(mod, pos.Filename)
+
+	return pos
+}
+
+func toRelativePath(mod *packages.Module, filename string) (string, bool) {
+	if filename == "" || !isCurrentModule(mod) {
+		return filename, false
+	}
+
+	rel, err := filepath.Rel(mod.Dir, filename)
+	if err != nil {
+		return filename, false
+	}
+
+	rel = filepath.ToSlash(rel)
+
+	if rel == ".." || strings.HasPrefix(rel, "../") {
+		return filename, false
+	}
+
+	return rel, true
+}
+
+func toAbsolutePath(mod *packages.Module, filename string) string {
+	if filename == "" || !isCurrentModule(mod) || filepath.IsAbs(filename) {
+		return filename
+	}
+
+	return filepath.Join(mod.Dir, filepath.FromSlash(filename))
+}
+
+func isCurrentModule(mod *packages.Module) bool {
+	return mod != nil && mod.Dir != "" && mod.Version == ""
+}
diff --git a/pkg/commands/run.go b/pkg/commands/run.go
index 913e736c3..e6c8779a8 100644
--- a/pkg/commands/run.go
+++ b/pkg/commands/run.go
@@ -643,0 +644 @@
+	b.WriteString("golangci-lint-cache/v2")
diff --git a/pkg/goanalysis/runners_cache.go b/pkg/goanalysis/runners_cache.go
index 9c6121f43..c576881c5 100644
--- a/pkg/goanalysis/runners_cache.go
+++ b/pkg/goanalysis/runners_cache.go
@@ -45 +45 @@
-						Pos:                  issue.Pos,
+						Pos:                  cache.RelativePosition(pkg.Module, issue.Pos),
@@ -117 +117 @@
-						Pos:                  issue.Pos,
+						Pos:                  cache.AbsolutionPosition(pkg.Module, issue.Pos),
